src/physics/negative_spalart_allmaras_rans_model.cpp

```diff
@@ -0,0 +1,718 @@
+template <int dim, int nstate, typename real>
+template<typename real2>
+real2 ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_Chi (
+    const real2 &nu_tilde,
+    const real2 &laminar_kinematic_viscosity) const
+{
+    // Compute coefficient Chi
+    const real2 Chi = nu_tilde/laminar_kinematic_viscosity;
+
+    return Chi;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template<typename real2>
+real2 ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_f_v1 (
+    const real2 &coefficient_Chi) const
+{
+    // Compute coefficient f_v1
+    real2 coefficient_f_v1;
+
+    if constexpr(std::is_same<real2,real>::value){ 
+        coefficient_f_v1 = coefficient_Chi*coefficient_Chi*coefficient_Chi/(coefficient_Chi*coefficient_Chi*coefficient_Chi+c_v1*c_v1*c_v1);
+    }
+    else if constexpr(std::is_same<real2,FadType>::value){ 
+        coefficient_f_v1 = coefficient_Chi*coefficient_Chi*coefficient_Chi/(coefficient_Chi*coefficient_Chi*coefficient_Chi+c_v1_fad*c_v1_fad*c_v1_fad);
+    }
+    else{
+        std::cout << "ERROR in physics/reynolds_averaged_navier_stokes.cpp --> compute_coefficient_f_v1(): real2 != real or FadType" << std::endl;
+        std::abort();
+    }
+
+    return coefficient_f_v1;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+real ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_f_v2 (
+    const real &coefficient_Chi) const
+{
+    // Compute coefficient f_v2
+    real coefficient_f_v2;
+    const real coefficient_f_v1 = compute_coefficient_f_v1(coefficient_Chi);
+
+    coefficient_f_v2 = 1.0-coefficient_Chi/(1.0+coefficient_Chi*coefficient_f_v1);
+
+    return coefficient_f_v2;
+}
+//----------------------------------------------------------------
+template <int dim, int nstate, typename real>
+template<typename real2>
+real2 ReynoldsAveragedNavierStokes_SAneg<dim,nstate,real>
+::compute_coefficient_f_n (
+    const real2 &nu_tilde,
+    const real2 &laminar_kinematic_viscosity) const
+{
+    // Compute coefficient f_n
+    real2 coefficient_f_n;
+    const real2 coefficient_Chi = compute_coefficient_Chi(nu_tilde,laminar_kinematic_viscosity);
+
+    if constexpr(std::is_same<real2,real>::value){ 
+        if (nu_tilde>=0.0)
+            coefficient_f_n = 1.0;
+        else
+            coefficient_f_n = (c_n1+coefficient_Chi*coefficient_Chi*coefficient_Chi)/(c_n1-coefficient_Chi*coefficient_Chi*coefficient_Chi);
+    }
+    else if constexpr(std::is_same<real2,FadType>::value){
+        const FadType const_one_fad = 1.0; 
+        if (nu_tilde>=0.0)
+            coefficient_f_n = const_one_fad;
+        else
+            coefficient_f_n = (c_n1_fad+coefficient_Chi*coefficient_Chi*coefficient_Chi)/(c_n1_fad-coefficient_Chi*coefficient_Chi*coefficient_Chi);
```

Comment:
coefficient_Chi*coefficient_Chi*coefficient_Chi is computed twice here; pre-compute it and use a variable